server/dbutil/dbutil.go

```diff
@@ -0,0 +1,161 @@
+package dbutil
+
+import (
+	"crypto/md5"
+	"database/sql"
+	"fmt"
+	"log"
+	"os"
+
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/pmezard/go-difflib/difflib"
+)
+
+const defaultExperimentID = 1
+
+// Bootstrap creates the necessary tables for the output DB. It is safe to call on a
+// DB that is already bootstrapped.
+func Bootstrap(db *sql.DB) error {
+	const (
+		createUsers = `CREATE TABLE IF NOT EXISTS users (
+			id INTEGER, github_username TEXT, auth TEXT, role INTEGER,
+			PRIMARY KEY (id))`
+		createExperiments = `CREATE TABLE IF NOT EXISTS experiments (
+			id INTEGER, name TEXT UNIQUE, description TEXT,
+			PRIMARY KEY (id))`
+		// TODO: consider a unique constrain to avoid importing identical pairs
+		createFilePairs = `CREATE TABLE IF NOT EXISTS file_pairs (
+			id INTEGER, name_a TEXT, name_b TEXT, hash_a TEXT, hash_b TEXT,
+			content_a TEXT, content_b TEXT, diff TEXT,experiment_id INTEGER,
+			PRIMARY KEY (id),
+			FOREIGN KEY(experiment_id) REFERENCES experiments(id))`
+		createAssignments = `CREATE TABLE IF NOT EXISTS assignments (
+			user_id INTEGER, pair_id INTEGER, experiment_id INTEGER,
+			answer INTEGER, duration INTEGER,
+			PRIMARY KEY (user_id, pair_id),
+			FOREIGN KEY (user_id) REFERENCES users(id),
+			FOREIGN KEY (pair_id) REFERENCES file_pairs(id),
+			FOREIGN KEY (experiment_id) REFERENCES experiments(id))`
+	)
+
+	if _, err := db.Exec(createUsers); err != nil {
```

Comment:
I would use a loop here. But up to you.

Reply:
What would be the idiomatic way in Go? How about this:

	tables := []string{createUsers, createExperiments, createFilePairs, createAssignments}

	for _, table := range tables {
		if _, err := db.Exec(table); err != nil {
			return err
		}
	}

Reply:
yes. I would write something like this to avoid multiple copy-past of `db.Exec; if err`.
There are many other ways how to write it, but I would do it as simple as possible. (your snippet above)